syscalls.c

```diff
@@ -942,12 +1080,17 @@ ssize_t syscalls_sys_read(char *ustack)
 	GETFROMSTACK(ustack, void *, buf, 1);
 	GETFROMSTACK(ustack, size_t, nbyte, 2);
 
+	if ((buf != NULL) && (nbyte != 0) && (vm_mapBelongs(proc->mapp, buf, nbyte, proc) < 0)) {
```

Comment:
I guess not exactly in scope of this PR but with unix socket `buf` is used without any additional checks, so NULL would still cause PF

Reply:
I only accept `buf == NULL` when `nbyte == 0`, `read()` should be resistant to such parameters (and I think it is indeed)

Reply:
Oh, ok, this condition is not ideal, will fix

Reply:
Yeah, this condition wasn't ideal, added a second one to patch it